lightning/src/ln/outbound_payment.rs

```diff
@@ -538,6 +570,8 @@ pub(super) struct SendAlongPathArgs<'a> {
 	pub session_priv_bytes: [u8; 32],
 }
 
+const BOLT_12_INVOICE_RETRY_STRATEGY: Retry = Retry::Attempts(3);
```

Comment:
Yeah, somehow it didn't occur to me that we can just put the `Retry` strategy in the new variants. We'll have to serialize it though. I assume we can't safely serialize `usize` as eight bytes since theoretically someone can give a `Retry::Attempts` with it greater than u32::MAX on a 64-bit machine, which can't be read on a 32-bit machine. Or should we just consider that undefined behavior?

Reply:
I guess my question really is should we change `Retry::Attempts` to wrap `u32` instead of `usize`?

Reply:
Added to this PR but can move it to the next one if you prefer.